kernel/kselftest.py

```diff
@@ -80,10 +80,27 @@ def setUp(self):
         if self.run_type == 'upstream':
             location = self.params.get('location', default='https://github.c'
                                        'om/torvalds/linux/archive/master.zip')
-            tarball = self.fetch_asset("kselftest.zip", locations=[location],
-                                       expire='1d')
-            archive.extract(tarball, self.workdir)
-            self.buldir = os.path.join(self.workdir, 'linux-master')
+            path = ''
+            git_location = False
```

Comment:
why do we need these?
```
if ".zip" in location or ".tar" in location:
```
wont work?

Reply:
yes  it won't work as tar location need to save in tar format otherwise it was easy to change

Reply:
makes sense, thanks for explanation.

Reply:
this error  will get 
```
020-07-30 01:22:04,555 process          L0689 INFO | Command 'rpm -q popt-devel' finished with 0 after 0.006664752960205078s
2020-07-30 01:22:04,597 parameters       L0143 DEBUG| PARAMS (key=location, path=*, default=https://github.com/torvalds/linux/archive/master.zip) => 'https://git.kernel.org/torvalds/t/linux-5.8-rc7.tar.gz'
2020-07-30 01:22:04,641 stacktrace       L0039 ERROR|
2020-07-30 01:22:04,641 stacktrace       L0042 ERROR| Reproduced traceback from: /usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/core/test.py:767
2020-07-30 01:22:04,641 stacktrace       L0045 ERROR| Traceback (most recent call last):
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|   File "/home/OpTest/avocado-fvt-wrapper/tests/avocado-misc-tests/kernel/kselftest.py", line 99, in setUp
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|     archive.extract(tarball, self.workdir)
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/utils/archive.py", line 286, in uncompress
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|     with ArchiveFile.open(filename) as x:
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/utils/archive.py", line 159, in open
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|     return cls(filename, mode)
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/utils/archive.py", line 137, in __init__
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|     self._engine = engine(self.filename, self.mode)
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|   File "/usr/lib64/python3.6/zipfile.py", line 1131, in __init__
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|     self._RealGetContents()
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|   File "/usr/lib64/python3.6/zipfile.py", line 1198, in _RealGetContents
2020-07-30 01:22:04,642 stacktrace       L0045 ERROR|     raise BadZipFile("File is not a zip file")
```

```
JOB ID     : d682f4e905506c718c35c71ff2b0abb678088445
JOB LOG    : /home/OpTest/avocado-fvt-wrapper/results/job-2020-07-30T01.21-d682f4e/job.log
 (1/1) avocado-misc-tests/kernel/kselftest.py:kselftest.test;power-upstream-8788: ERROR: File is not a zip file (2.36 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/OpTest/avocado-fvt-wrapper/results/job-2020-07-30T01.21-d682f4e/results.html
JOB TIME   : 2.72 s
[root@system]# 
````

Reply:
ah i think now you unsrstand why i am saying it was bug  and want in to commits , sorry if i was not clear